File: src/main/java/com/junchen/back/service/AmAddressbookService.java
package com.junchen.back.service;
import java.util.List;

import com.junchen.back.pojo.ErpAmAddressbook;
import com.junchen.back.myentity.PageResult;
/**
 * amAddressbook服务层接口
 * @author Administrator
 *
 */
public interface AmAddressbookService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<ErpAmAddressbook> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(ErpAmAddressbook amAddressbook);
	
	
	/**
	 * 修改
	 */
	public void update(ErpAmAddressbook amAddressbook);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public ErpAmAddressbook findOne(Long id);
	
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(ErpAmAddressbook amAddressbook, int pageNum,int pageSize);


	public List<String> findCompany();
	
}

File: src/main/webapp/js/service/amAddressbookService.js
//amAddressbook服务层
app.service('amAddressbookService',function($http){
	    	
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('../amAddressbook/findAll.do');		
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('../amAddressbook/findPage.do?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('../amAddressbook/findOne.do?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('../amAddressbook/add.do',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('../amAddressbook/update.do',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('../amAddressbook/delete.do?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('../amAddressbook/search.do?page='+page+"&rows="+rows, searchEntity);
	}   
	//查找公司下拉列表
	this.findCompany = function(){
		return $http.get('../amAddressbook/findCompany.do');
	}
	
});
